Client/mods/deathmatch/logic/luadefs/CLuaResourceDefs.cpp

```diff
@@ -220,34 +222,30 @@ int CLuaResourceDefs::GetResourceConfig(lua_State* luaVM)
     return 1;
 }
 
-int CLuaResourceDefs::GetResourceName(lua_State* luaVM)
+std::string CLuaResourceDefs::GetResourceName(lua_State* luaVM, std::optional<CResource*> resourceElement)
 {
-    // Verify arguments
-    CResource*       pResource = NULL;
-    CScriptArgReader argStream(luaVM);
-    argStream.ReadUserData(pResource);
-
-    if (!argStream.HasErrors())
+    if (!resourceElement)
     {
-        if (pResource)
-        {
-            // Grab its name and return it
-            const char* szName = pResource->GetName();
-            if (szName)
-            {
-                lua_pushstring(luaVM, szName);
-                return 1;
-            }
-        }
-        else
-            m_pScriptDebugging->LogBadPointer(luaVM, "resource", 1);
+        CLuaMain*  localVM = m_pLuaManager->GetVirtualMachine(luaVM);
+        CResource* localResource;
+
+        if (!localVM)
+            throw std::runtime_error("Couldn't find the virtual machine");
+
+        localResource = localVM->GetResource();
+
+        if (!localResource)
+            throw std::runtime_error("Couldn't find the resource");
+
+        resourceElement = localResource;
     }
-    else
-        m_pScriptDebugging->LogCustom(luaVM, argStream.GetFullErrorMessage());
 
-    // Failed
-    lua_pushboolean(luaVM, false);
-    return 1;
+    std::string resourceName = (*resourceElement)->GetName();
+
+    if (resourceName.empty())
+        throw std::runtime_error("The resource name is empty");
```

Comment:
There's no need in this check because you simply cannot create a resource with empty name in MTA.